command/meta.go

```diff
@@ -480,8 +485,33 @@ func (m *Meta) showDiagnostics(vals ...interface{}) {
 	diags = diags.Append(vals...)
 	diags.Sort()
 
+	if len(diags) == 0 {
+		return
+	}
+
+	diags = diags.ConsolidateWarnings(1)
+
 	// Since warning messages are generally competing
-	diags = diags.ConsolidateWarnings()
+	if m.compactWarnings {
+		// If the user selected compact warnings and all of the diagnostics are
+		// warnings then we'll use a more compact representation of the warnings
+		// that only includes their summaries.
+		// We show full warnings if there are also errors, because a warning
+		// can sometimes serve as good context for a subsequent error.
+		useCompact := true
+		for _, diag := range diags {
```

Comment:
If we only want to upgrade the output on errors, couldn't this just be `HasErrors()`?

Reply:
This is some future-proofing for a possible future scenario where there is a third diagnostic severity. My goal here is to explicitly check that the entire diagnostic set is warnings, as required by the contract of `DiagnosticWarningsCompact`, rather than to indirectly check it by assuming that error is the only non-warning severity.

(If this comes up again, perhaps we can add an `OnlyWarnings` accessor to `Diagnostics` to encapsulate this, but so far I don't recall us needing this check anywhere else.)